radicle-cli/src/commands/track.rs

```diff
@@ -0,0 +1,94 @@
+use std::ffi::OsString;
+
+use anyhow::anyhow;
+
+use radicle::identity::project::Id;
+use radicle::node::Handle;
+
+use crate::terminal as term;
+use crate::terminal::args::{Args, Error, Help};
+
+pub const HELP: Help = Help {
+    name: "track",
+    description: "Manage radicle project tracking relationships",
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad track <id> [--fetch]
+
+Options
+
+    --fetch                Fetch the peer's refs into the working copy
+    --verbose, -v          Verbose output
+    --help                 Print help
+"#,
+};
+
+#[derive(Debug)]
+pub struct Options {
+    pub id: Id,
+    pub fetch: bool,
+    pub verbose: bool,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut id: Option<Id> = None;
+        let mut fetch = true;
+        let mut verbose = false;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("no-fetch") => fetch = false,
+                Long("verbose") | Short('v') => verbose = true,
+                Value(val) if id.is_none() => {
+                    let val = val.to_string_lossy();
+
+                    if let Ok(val) = Id::from_human(&val) {
+                        id = Some(val);
+                    } else {
+                        return Err(anyhow!("invalid ID '{}'", val));
+                    }
+                }
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                _ => {
+                    return Err(anyhow!(arg.unexpected()));
+                }
+            }
+        }
+
+        let id = id.ok_or_else(|| anyhow!("ID is missing, usage: `rad track <id>`"))?;
+
+        Ok((Options { id, fetch, verbose }, vec![]))
+    }
+}
+
+pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let profile = ctx.profile()?;
+    let node = radicle::node::connect(&profile.node())?;
+
+    term::info!(
+        "Establishing 🌱 tracking relationship for {}",
+        term::format::highlight(&options.id.to_human())
```

Comment:
is `id.to_human()` good enough? I'd prefer if we display something about repo as well, e.g. name like before.

Reply:
I switched to `payload.name` instead https://github.com/radicle-dev/heartwood/compare/5d50941d85946a65945fe7cb4a83cd69856870fd..460dee1f002c8887f616a7aa74d2431c280e1b42